main.tf

```
@@ -49,8 +49,8 @@ resource "aws_s3_bucket" "this" {
   }
 
   logging {
-    target_bucket = var.bucket
-    target_prefix = "TFStateLogs/"
+    target_bucket = var.logging_target_bucket != "" ? var.logging_target_bucket : var.bucket
```

I quite like the `coalesce` pattern in this case

```suggestion
    target_bucket = coalesce(var.logging_target_bucket, var.bucket)
```